### ISSUE TITLE
Installing extra packages into pipeline environment breaks datamodels

### ISSUE THREAD

**Issue by PaddyKavanagh** [spacetelescope/jwst#7334]
Hi,
When I try to install additional packages into the pipeline environment (e.g., jupyter, specutils, etc.), the version of certifi is updated, which then causes an error when trying to load a file into a datamodel, e.g.:

FileNotFoundError: Unable to fetch schema from non-file URL: http://stsci.edu/schemas/jwst_datamodel/ifucube.schema

Rolling back the certifi version fixes the problem:

pip install certifi==2022.5.18.1

but it's very annoying to have to do this after every package install. I'm using dev versions of pipeline but have heard the same problem from others using jwst1.8.0

**Comment by jdavies-st** [comment]
So what dependency of `jwst` is rolling back `certifi`?  Could you show via step-by-step workflow how this happens so it can reproduced?

**Comment by PaddyKavanagh** [comment]
James, apologies for the delay in replying. After testing again with the most recent dev version I cannot seem to reproduce the error! So I don't know what has changed or why it works fine now but this had been an issue for me and others. Perhaps this ticket can be closed but I will open it again of the issue resurfaces.

**Comment by jdavies-st** [comment]
Thanks @PaddyKavanagh.  I'm mistaken, as it looks like it is being pinned in `jwst` currently:

https://github.com/spacetelescope/jwst/pull/6927

https://github.com/spacetelescope/jwst/blob/1b8d288cf8be4c1b0490119e5ef13c65761acfe2/setup.cfg#L33

Unexpected!

@zacharyburnett and @hbushouse is there a reason to be pinning `certifi` in `setup.cfg`?  I read the linked issues above, and it seems to be an issue with the interaction of `pip` and the conda package of `certifi` when one creates a bare conda env with only Python installed.  `certifi` comes along for the ride, and it is the conda package of certifi.  It is the fact that package is installed by conda and not pip that makes the reporting in `pip freeze` broken.  A solution would be to reinstall it via pip.

Generally, pinning in a library such as `jwst` is not a good idea, especially since `jwst` doesn't even directly depend on `certifi` (`requests` pulls it in).  Better would be to pin it as needed in `requirements-sdp.txt` if it is needed for DMS/SDP processing.

But of course the version is not the problem, it's the behavior of pip with the conda packager.  I.e. the same malformed `pip freeze` happens with all versions of `certifi`.   `setup.cfg`, we should probably be using something like the following in JenkinsfileRT.

```
"pip install certifi -U --force-reinstall",
"pip install -e .[test,sdp] --no-cache-dir", 
"pip install pytest-xdist", 
"pip freeze", 
```

instead of

https://github.com/spacetelescope/jwst/blob/1b8d288cf8be4c1b0490119e5ef13c65761acfe2/JenkinsfileRT#L79-L81

This will guarantee the latest version of `certifi` is used (good) and that it is installed by pip, so that `pip freeze` works as expected.

**Comment by zacharyburnett** [comment]
@jdavies-st good catch, I was baffled why this issue was occurring but it being an issue with `conda` vs `pip` build makes sense

**Comment by jdavies-st** [comment]
Resolved by #7355.
